common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -596,6 +597,40 @@ private void validateAuthScheme(@NonNull final MicrosoftStsTokenRequest request,
         }
     }
 
+    /**
+     * Validates that the token response contains an access token, id_token and refresh token
+     *
+     *      * @param response The idp response.
+     * @throws ClientException
+     */
+    private void validateTokensAreInResponse(@NonNull final MicrosoftStsTokenResponse response)
+            throws ClientException {
+
+        String clientException = null;
+        String description = null;
+
+        if(response.getIdToken() == null){
```

Comment:
you probably want to check for empty strings too (here and also for at, rt): `TextUtils.isEmpty(s)`

Reply:
Makes sense.